lwm2m is unable to send further messages after send encountered certain signals

dominik-dess opened the pull request:
Fixed issue that caused message to be not correctly reset even after it is consumed after send sets errno to EAGAIN or EWOULDBLOCK

We encountered the issue while working with a nRF9160 based project.
In certain situation we encountered the device being unable to send lwm2m messages. On debugging we found that "socket_send_message" is never reached in the socket_loop function.
This bug was found together with the developer that added the line as well.

rlubos commented on subsys/net/lib/lwm2m/lwm2m_engine.c:
```diff
@@ -1050,7 +1050,6 @@ static int lwm2m_send_message(struct lwm2m_message *msg)
 	if (rc < 0) {
 		if (errno == EAGAIN || errno == EWOULDBLOCK) {
 			LOG_WRN("Failed to send packet, would block");
-			return -errno;
```
In that case, we should perhaps drop the entire `if (errno == EAGAIN || errno == EWOULDBLOCK) {` condition? The error will be logged just below.

dominik-dess-gcx replied:
Good point, I removed this condition

rlubos reviewed (commented):
Something went wrong with the commit message - please set a proper title (with `net: lwm2m: ` prefix), signoff at the bottom.

huelsenfruchtzwerg commented:
As "the developer that added the bug" :sweat_smile: : Originally the code was intended to not remove the message from the send queue when send report EAGAIN or EWOULDBLOCK, but we changed this during the review process, but I overlooked to then clear the message also for those error conditions, effectively leaking the messag.

Commit:
net: lwm2m: lwm2m stops sending messages after encountering signals on send

Fixed issue that caused message to be not correctly reset even after
 it is consumed after send sets errno to EAGAIN or EWOULDBLOCK

Signed-off-by: Dominik Dess <dominik.dess@grandcentrix.net>

rlubos commented:
The commit guidelines are pretty strict in Zephyr, please address the following issues:

```
1: UC5 Title exceeds max length (142>75): "net: lwm2m: Fixed issue that caused message to be not correctly reset even after it is consumed after send sets errno to EAGAIN or EWOULDBLOCK"
1: UC6 Body has no content, should at least have 1 line.
```

dominik-dess-gcx commented:
Really sorry about the malformed commit messages, thanks for pointing that out
